src/vma/dev/allocator.cpp

```diff
@@ -94,7 +94,8 @@ void* vma_allocator::alloc_and_reg_mr(size_t size, ib_ctx_handler *p_ib_ctx_h)
 	// fallthrough
 	case ALLOC_TYPE_CONTIG:
 #ifdef VMA_IBV_ACCESS_ALLOCATE_MR
-		if (!safe_mce_sys().is_hypervisor && g_p_ib_ctx_handler_collection->get_ib_cxt_list()) {
+		if (mce_sys_var::HYPER_MSHV != safe_mce_sys().hypervisor &&
+				g_p_ib_ctx_handler_collection->get_ib_cxt_list()) {
```

Comment:
Do we still need the check of g_p_ib_ctx_handler_collection->get_ib_cxt_list() ?
if not - please remove it

Reply:
done